File: day-1/puzzle-1.js
const values = require("./input.json");

function puzzle1 ({
  values
}) {
  return values.reduce(
    (sum, value) => {
      const numbersFound = value.match(/[0-9]/gi);

      if (!numbersFound.length) return sum;

      const numberForValue = +`${numbersFound[0]}${numbersFound[numbersFound.length - 1]}`;

      return numberForValue + sum;
    },
    0
  );
}

// const testValues = [
//   "1abc2",
//   "pqr3stu8vwx",
//   "a1b2c3d4e5f",
//   "treb7uchet",
// ]

console.log(puzzle1({
  values
}));

File: day-1/puzzle_1.exs
{ :ok, file_content } = File.read("./day-1/input.json")
{ :ok, input } = Jason.decode(file_content)

puzzle_1 = fn input when is_list(input) ->
  Enum.reduce(
    input, 0, fn value, sum ->
      calc = Regex.scan ~r/[0-9]/, value
      sum + String.to_integer "#{List.first(calc)}#{List.last(calc)}", 10
    end
  )
end

puzzle_1.(input) |> IO.puts
